crates/biome_analyze/src/lib.rs

```diff
@@ -843,32 +851,19 @@ impl<'analysis> AnalysisFilter<'analysis> {
                 enabled_rules.iter().any(|filter| filter.match_group::<G>())
             })
             && self.disabled_rules.map_or(true, |disabled_rules| {
-                !disabled_rules
-                    .iter()
-                    .any(|filter| filter.match_group::<G>())
+                !disabled_rules.iter().any(|filter| {
+                    matches!(filter, RuleFilter::Group(_)) && filter.match_group::<G>()
+                })
             })
     }
 
     /// Return `true` if the rule `R` matches this filter
-    pub fn match_rule<R>(&self) -> bool
-    where
-        R: Rule,
-    {
-        self.match_group::<R::Group>()
-            && self.enabled_rules.map_or(true, |enabled_rules| {
-                enabled_rules.iter().any(|filter| filter.match_rule::<R>())
-            })
-            && self.disabled_rules.map_or(true, |disabled_rules| {
-                !disabled_rules.iter().any(|filter| filter.match_rule::<R>())
-            })
-    }
-
-    /// It creates a new filter with the set of [enabled rules](RuleFilter) passed as argument
-    pub fn from_enabled_rules(enabled_rules: Option<&'analysis [RuleFilter<'analysis>]>) -> Self {
-        Self {
-            enabled_rules,
-            ..AnalysisFilter::default()
-        }
+    pub fn match_rule<R: Rule>(&self) -> bool {
+        self.enabled_rules.map_or(true, |enabled_rules| {
```

Comment:
We don't need to call `match_group` because this check is already handled in `filter.match_rule`.